grails-app/controllers/org/pih/warehouse/inventory/StockTransferController.groovy

```diff
@@ -99,18 +97,26 @@ class StockTransferController {
 
         try {
             stockTransferService.deleteStockTransfer(params.orderId ?: params.id)
-        } catch (IllegalArgumentException e) {
-            flash.message = "${warehouse.message(code: 'default.not.found.message', args: [warehouse.message(code: 'inventory.stockTransfer.label', default: 'Stock Transfer'), params.id])}"
+            flash.message = g.message(
+                    code: 'react.stockMovement.deleted.success.message.label',
+                    body: 'Stock Movement has been deleted successfully',
+            )
+        } catch (Exception e) {
+            flash.message = "${warehouse.message(code: 'stockMovement.delete.error.message')}"
+            redirect(controller: "stockMovement", action: "show", id: params.id)
+            return
         }
 
+        params = params << ["flash": flash as JSON]
```

Comment:
I think this `params << ["flash": flash as JSON]` should happen in the `list` action or in the `list` action of the stock movement controller before rendering the template.